src/TapeChart.js

```diff
@@ -0,0 +1,48 @@
+class TapeChart {
+  constructor(users, rooms, bookings) {
+    this.users = users;
+    this.rooms = rooms;
+    this.bookings = bookings;
+  }
+
+getAvailableRoomsByDate(date) {
+    let availableByDate = this.rooms.reduce((acc, room) => {
+      let bookingsByRoom = this.bookings.filter(booking =>
+        booking.roomNumber === room.number);
+      if (!bookingsByRoom.some(booking => booking.date === date)) {
+        acc.push(room);
+      }
+        return acc;
+    }, [])
+  return availableByDate;
+}
+
+getOccupancy(date) {
+  return 100 - (this.getAvailableRoomsByDate(date).length
+    / this.rooms.length * 100);
+}
+
+getBookingsByDate(date) {
+  return this.bookings.filter(booking => booking.date === date);
+}
+
+getDaysRevenue(date) {
+  let bookingsByDate = this.getBookingsByDate(date);
+  return bookingsByDate.reduce((acc, booking) => {
+    this.rooms.forEach(room => {
+      if (room.number === booking.roomNumber) {
```

Comment:
parseInt(booking.roomNumber) would help just in case there are pesky string getting into the booking dat.

Reply:
Good looking out.  I've wrapped things in Number rather than parseInt() and gotten the same effect.   I need to research the difference between these two.  I think I'll stick with Number to start since I'm more familiar with it.

Reply:
I think parseInt returns a whole number, so .9 would become 1.